### PR TITLE
Add Windows Nano Server 1803 Dockerfiles

### PR DESCRIPTION
**Pull request by natemcmaster**
Add Nano Server images for Windows 1803 (rs4)

cc @muratg @glennc 

Closes https://github.com/aspnet/aspnet-docker/issues/416

### CODE REVIEW

**Comment by natemcmaster** [comment]
@dotnet-bot test ci please

File: build-pipeline/windows-pipeline.groovy [hunk 1]
```diff
@@ -18,6 +18,9 @@ def buildImages = {
 if (params.os == '1709') {
     simpleNode('windows.10.amd64.serverrs3.open', buildImages)
 }
+else if (params.os == '1803') {
+    simpleNode('windows.10.amd64.serverrs4.open', buildImages)
```
Comment (MichaelSimons) [inline build-pipeline/windows-pipeline.groovy]: Does this exist?  I thought it was still being created - https://github.com/dotnet/core-eng/issues/3365

File: README.aspnetcore-build.md [hunk 2]
```diff
@@ -19,7 +23,7 @@ This repository contains images that are used to compile/publish ASP.NET Core ap
 - [`1.1.8-1.1.9-nanoserver-sac2016`, `1.1.8-1.1.9`, `1.1`, `1` (*1.1/nanoserver-sac2016/sdk/Dockerfile*)](https://github.com/aspnet/aspnet-docker/blob/master/1.1/nanoserver-sac2016/sdk/Dockerfile)
 - [`2.0.8-2.1.200-nanoserver-sac2016`, `2.0-nanoserver-sac2016`, `2.0.8-2.1.200`, `2.0`, `2`, `latest` (*2.0/nanoserver-sac2016/sdk/Dockerfile*)](https://github.com/aspnet/aspnet-docker/blob/master/2.0/nanoserver-sac2016/sdk/Dockerfile)
 
->**Note:** ASP.NET Core multi-arch tags, such as 2.0, have been updated to use nanoserver-1709 images if your host is Windows Server 2016 Version 1709 or higher or Windows 10 Fall Creators Update (Version 1709) or higher. You need Docker 17.10 or later to take advantage of these updated tags.
+>**Note:** ASP.NET Core multi-arch tags, such as 2.0, have been updated to use nanoserver-1803 images if your host is Windows Server 2016 Version 1803 or higher or Windows 10 Fall Creators Update (Version 1803) or higher. You need Docker 17.10 or later to take advantage of these updated tags.
```
Comment (MichaelSimons) [inline README.aspnetcore-build.md]: This note is kind off odd now because it still applies to 1709.  We dropped this from microsoft/dotnet.

Comment (natemcmaster) [reply README.aspnetcore-build.md]: I'll remove it

File: build-pipeline/windows-pipeline.groovy (hunk 1 above)
Comment (natemcmaster) [reply build-pipeline/windows-pipeline.groovy]: Is this the right way to identify the RS4 nodes? 

I thought I saw comments that this was ready to go: https://github.com/dotnet/core-eng/issues/3365#issuecomment-389995754.

**Comment by natemcmaster** [comment]
Merging now to get official builds running. I'll update the Jenkins config if necessary.

**Pull request merged**